### ISSUE TITLE
Pros/Cons of this libary vs tsoa

### ISSUE THREAD

**Issue by dgreene1** [thiagobustamante/typescript-rest-swagger#84]
Hello. I noticed that this library has a very large overlap in goals with [tsoa](https://github.com/lukeautry/tsoa). I'm curious if:
(a) someone can describe some of the offerings that this library has over tsoa and
(b) if the maintainers of this library would be interested in combining (or something like that) so that we could have more strength in our union.

Full transparency: I recently volunteered to be a maintainer of tsoa. To be clear on a few points:
(a) I'm not _the_ maintainer of tsoa. I'm just a guy who liked using it a lot and wanted to support it
(b) This is really just a conversation. I'd like to see TypeScript become one of the more popular server-side languages, and I think having a really strong unified approach to developing services would help that goal. That being said, I obviously think that providing users with options is also good to. Which, in summary, is why I'm looking to understand the comparative offers to understand why someone would choose this library over tsoa.

**Comment by dgreene1** [comment]
Update: I've done a little more research into the two libraries and I will do my best to update this document. If I've left anything out on either side, please comment below.

**Comment by YousefED** [comment]
Thanks! Useful. I also noticed there's some overlap between tsoa / trs in terms of JSON schema generation from Typescript with existing libraries. I've done some early work on this @ https://github.com/yousefed/typescript-json-schema (now mostly superseded by https://github.com/vega/ts-json-schema-generator), figured it might be useful to re-use

**Comment by ArminEberle** [comment]
Thanks also from my side. We recently found tsoa and are now migrating to it. 
A really decent thing that matters to us is actually something that comes as a hidden benefit when using tsoa is the fact that at runtime, we don't need the decorator metadata anymore:
We're packaging our server code with esbuild so when can easily ship the code into a docker-image.
With ts-rest, we have to compile the code first with typescript tsc and build from the generated .js sources, as esbuild (and webpack as well) cannot handle decorator code emit when compiling from .ts .
With tsoa, I hope that we can run esbuild directly on the .ts files and and run typescript tsc only for typechecking, in parallel to esbuild. This will save us a minute or so on every build, which does not really matter on the build server, but locally during development, this will save the developer something in the range of 5-10 minutes per day.
